File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/LinkedListNodeHandler.java
package combined.avro_events.consumer;

import com.example.events.LinkedListNode;
import combined.avro_events.header.StandardHeaders;
import java.lang.Void;
import java.util.concurrent.CompletableFuture;

/** Handler interface for linked-list-node topic events */
public interface LinkedListNodeHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(
    String key,
    LinkedListNode value,
    StandardHeaders headers
  );
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/LinkedListNodeHandler.kt
package combined.avro_events.consumer

import com.example.events.LinkedListNode
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Uni

/** Handler interface for linked-list-node topic events */
interface LinkedListNodeHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: LinkedListNode,
    headers: StandardHeaders
  ): Uni<Unit>
}